src/clr-angular/data/datagrid/datagrid-page-size.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, Input } from '@angular/core';
+import { Page } from './providers/page';
+
+@Component({
+  selector: 'clr-dg-page-size',
+  template: `
+    <ng-content></ng-content>
+    <div class="clr-select-wrapper">
+      <select [(ngModel)]="page.size">
+        <option *ngFor="let option of pageSizeOptions" [ngValue]="option">{{option}}</option>
+      </select>
+    </div>
+  `,
+})
+export class ClrDatagridPageSize {
+  private _pageSizeOptions: number[] = [];
```

Comment:
Does it make sense to offer default pageSizeOptions so users don't _have_ to supply them if they want the easy way out?

Reply:
If they don't want to have page size options, they can opt to not use it in their markup. 

With:
```
    <clr-dg-pagination #pagination [clrDgPageSize]="20">
      <clr-dg-page-size [clrPageSizeOptions]="[10,20,50,100]">Users per page</clr-dg-page-size>
      {{pagination.totalItems}} users
    </clr-dg-pagination>
```
Without:
```
    <clr-dg-pagination #pagination [clrDgPageSize]="20">
      {{pagination.totalItems}} users
    </clr-dg-pagination>
```

I would assume that they would want to supply them if they are choosing to use the component. What do you think?

Reply:
So if the array is empty the select box doesn't show up?

Reply:
Yeah it shows up but is blank. If you think it's worthwhile, I can default it to include one option with the value that is passed by `[clrDgPageSize]`?

Reply:
Why would we want it to show up blank? 

If you had the pagination select box when the array is empty, then I think it's fine the way it is. Basically that's a way for people to turn off that feature if it's not useful for them (and by default it's turned off).

But if we are going to show the select box no matter what, we should have a set of default options. Which is problematic if our defaults are larger than their dataset.

Reply:
I'll set it so that it shows up with one option of value passed in the `clrDgPageSize`. I'm afraid that if we hide the select when array is empty, it'll just cause more confusion.

Reply:
I agree with the behavior Jeeyun has on this one. Components with a "mandatory" input are not hard to grasp, the basic example in every tutorial is a "greeter" component that utterly fails if you don't give it a `name` input.

I like the gracious fallback to a select that only has a single option, the current page size. So 👍 on that.
To address Scott's concern, see my next comment.